### PR TITLE
Use s3 to host the preview environments

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
Use S3 to host the preview environment instead of gh-pages because it has rate limits.

This work is similar to what we did for docs and mattermost-developer-documentations repos.

for upstream PRs the preview will run and build, for fork PRs we will need to add the `preview-environment` label to trigger the build

#### Ticket Link
JIRA: https://mattermost.atlassian.net/browse/DOPS-522

### CODE REVIEW

**Review by nevyangelova — DISMISSED** [review]
Thank you :tada: :tada: :tada: :tada: :tada:

File: .github/workflows/preview-env-fork.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+          destination_dir: ${{ steps.pr.outputs.id }}
+
+      - name: Add comment to PR
+        uses: peter-evans/create-or-update-comment@v1
+        if: ${{ success() &&  github.event.workflow_run.conclusion == 'success' && steps.label-checker.outputs.result == 'true' }}
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          issue-number: ${{ steps.pr.outputs.id }}
+          body: |
+            Newest code from ${{ github.actor }} has been published to [preview environment](http://http://mattermost-compass-preview-pull.s3-website-us-east-1.amazonaws.com/${{ steps.pr.outputs.id }}) for Git SHA ${{ github.event.workflow_run.head_sha }}
```
Comment (metanerd) [inline .github/workflows/preview-env-fork.yml:102]:
```suggestion
            Newest code from ${{ github.actor }} has been published to [preview environment](http://mattermost-compass-preview-pull.s3-website-us-east-1.amazonaws.com/${{ steps.pr.outputs.id }}) for Git SHA ${{ github.event.workflow_run.head_sha }}
```

**Review by metanerd — APPROVED** [review]
Thank you so much!

**Review by michelengelen — APPROVED** [review]
🎉 🎉 🎉 🎉 🎉 🎉

**Review by nevyangelova — APPROVED** [review]

**Pull request merged**